renku/service/views/templates.py

```diff
@@ -65,3 +75,84 @@ def read_manifest_from_template(user, cache):
     return result_response(
         ManifestTemplatesResponseRPC(), {'templates': manifest}
     )
+
+
+@use_kwargs(ProjectTemplateRequest)
+@marshal_with(ProjectTemplateRequest)
+@header_doc(
+    'Create a new project starting from a target template available in a '
+    'remote repositpry.',
+    tags=(TEMPLATES_BLUEPRINT_TAG, )
+)
+@templates_blueprint.route(
+    '/templates.create_project',
+    methods=['POST'],
+    provide_automatic_options=False,
+)
+@handle_base_except
+@handle_git_except
+@handle_renku_except
+@handle_validation_except
+@handle_schema_except
+@accepts_json
+@requires_cache
+@requires_identity
+def create_project_from_template(user, cache):
+    """Create a new project starting form target template."""
+    ctx = ProjectTemplateRequest().load({
+        **user,
+        **request.json,
+    },
+                                        unknown=EXCLUDE)
+
+    # Clone project and find target template
+    template_project = _project_clone(user, ctx)
+    templates = read_template_manifest(template_project.abs_path)
+    template = next((
+        template
+        for template in templates if template['folder'] == ctx['identifier']
+    ), None)
+    if template is None:
+        return error_response(
+            INVALID_PARAMS_ERROR_CODE,
+            'invalid identifier for target repository'
+        )
+
+    # Verify missing parameters
+    template_parameters = template.get('variables', {})
```

Comment:
in https://github.com/SwissDataScienceCenter/renku-python/pull/1272 this is cone in `renku/cli/init.py`, it'd be nice to have a single method for missing/ignored parameter check that can be used both in the service and the cli.

Reply:
good point! Now that #1272 is merged, I have no excuses for not doing this :slightly_smiling_face:

Reply:
I double-checked the code, and I think it makes sense to keep them separate (I slightly updated it as you suggested below). The logic is similar, but the input parameters are formatted in a different way (`{ key: <keyname>, value: <value> }` for API vs `{ <keyname>: <value> }` for cli) and missing one triggers an error in the API but does something different in the cli, so the common code is very limited.

I can still create a single method if you think that helps in reading the code better.

Reply:
Ok I tried to use the `repo_sync` function but I had to change a detail since it's not possible to fetch from a non-existing repo before pushing. I hope that change makes sense.